packages/kit/src/exports/vite/static_analysis.js

```diff
@@ -0,0 +1,175 @@
+import { tsPlugin } from '@sveltejs/acorn-typescript';
+import { Parser } from 'acorn';
+import { read } from '../../utils/filesystem.js';
+
+const dynamic_page_options = new RegExp(`${['entries', 'config'].join('|')}`);
+
+const statically_analysable_page_options = new RegExp(
+	`${['prerender', 'csr', 'ssr', 'trailingSlash'].join('|')}`
+);
+
+const parser = Parser.extend(tsPlugin());
+
+/**
+ * Collects exported page options from a +page.js/+layout.js file.
+ * Returns `null` if any export cannot be statically analyzed.
+ * @param {string} node_path
+ */
+function statically_analyse_exports(node_path) {
+	const input = read(node_path);
+
+	// if there's something we can't analyse, we mark the node as not statically analysable
+	if (dynamic_page_options.test(input)) {
+		return null;
+	}
+
+	// if there's nothing to analyse, we can skip parsing
+	if (!statically_analysable_page_options.test(input)) {
```

Comment:
We gotta check for `export * from` not being part of the input (and if we remove the `dynamic_page_options` test above, then we gotta check for those strings not occuring, too)

Reply:
Thanks for checking. I've changed it so that we only return early if there are no page options (ssr, csr, prerender, trailingSlash, config, entries) and no export all declaration.